### PR TITLE
[enhancement](Nereids) New optimizer support check column privileges

### PR DESCRIPTION
**Pull request by 924060929**
## Proposed changes

New optimizer support
1. check column privileges for table/view.
1. combine check table privilege and column privilege to `CheckPrivileges` rule
2. forbidden skip `CheckPrivileges` and `CheckRowPolicy`

### CODE REVIEW

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateSortUnderSubqueryOrView.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+public class EliminateSortUnderSubqueryOrView implements RewriteRuleFactory {
+    @Override
+    public List<Rule> buildRules() {
+        return ImmutableList.of(
+            RuleType.ELIMINATE_ORDER_BY_UNDER_SUBQUERY.build(
+                logicalSubQueryAlias(logicalSort())
+                        .then(subq -> subq.withChildren(subq.child().child(0)))
+            ),
+            RuleType.ELIMINATE_ORDER_BY_UNDER_SUBQUERY.build(
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateSortUnderSubqueryOrView.java:38]:
```suggestion
            RuleType.ELIMINATE_ORDER_BY_UNDER_VIEW.build(
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalView.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+/** LogicalView */
+public class LogicalView<BODY extends Plan> extends LogicalUnary<BODY> {
+    private final View view;
+
+    /** LogicalView */
+    public LogicalView(View view, BODY body) {
+        super(PlanType.LOGICAL_VIEW, Optional.empty(), Optional.empty(), body);
+        this.view = Objects.requireNonNull(view, "catalog can not be null");
+        Preconditions.checkArgument(body instanceof LogicalPlan);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalView.java:49]: add check msg

**Review by morrySnow — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalView.java (hunk 2 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalView.java:49]: next pr will fix this problem

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateSortUnderSubqueryOrView.java (hunk 1 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateSortUnderSubqueryOrView.java:38]: next pr will fix this problem

**Review by yiguolei — APPROVED** [review]

**Pull request merged**